### PR TITLE
[WIP] Hydrate the database module with the Term class.

### PR DESCRIPTION
**Pull request by dsagang**
#30

### CODE REVIEW

**Comment by allenbrubaker** [comment]
#30 states that we need to add behavioral change for converting tags from an array to a comma delimited list of strings and vice-versa when storing and retrieving from database.

File: src/database.js [hunk 1]
```diff
@@ -80,7 +81,15 @@ export default class Database{
                 }
 
                 done();
-                callback(result);
+                result.rows.forEach(function(element){
```
Comment (williambogaiv) [inline src/database.js]: @dsagang, double-check your logic here. Using `forEach` does not overwrite the original array. You might be looking for something like [map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map).

File: src/database.js [hunk 2]
```diff
@@ -39,7 +39,8 @@ export default class Database{
                                   }
 
                                   done();
-                                  callback(result.rows[0]);
+                                  var term = new Term(result.rows[0].id, result.rows[0].term, result.rows[0].definition, result.rows[0].tags.join([separator = ',']) || []);
```
Comment (williambogaiv) [inline src/database.js]: @dsagang, remember when readings docs. like https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split, the brackets are not part of the actual implementation. See some of the examples on the page on how to implement.

**Comment by williambogaiv** [comment]
It looks like this functionality has already been introduced with other PRs. Any reason to keep this open? If not, @dsagang, just close this PR.

**Pull request closed without merging**